### PR TITLE
bpo-33089: Add math.dist() for computing the Euclidean distance between two points

### PR DESCRIPTION
**Pull request by rhettinger**
https://bugs.python.org/issue33089

### CODE REVIEW

File: Modules/mathmodule.c [hunk 1]
```diff
@@ -2031,6 +2031,87 @@ math_fmod_impl(PyObject *module, double x, double y)
...
+    m = PyTuple_GET_SIZE(p);
+    n = PyTuple_GET_SIZE(q);
+    if (m != n) {
+        PyErr_SetString(PyExc_ValueError,
+                        "both points must have the same number of dimensions");
+        return NULL;
+
+    }
+    diffs = (double *) PyObject_Malloc(n * sizeof(double));
+    if (diffs == NULL)
```
Comment (sir-sigurd) [inline Modules/mathmodule.c]: Braces are required here (PEP7).

File: Modules/mathmodule.c [hunk 2]
```diff
@@ -2031,6 +2031,87 @@ math_fmod_impl(PyObject *module, double x, double y)
...
+        return NULL;
+    for (i=0 ; i<n ; i++) {
```
Comment (sir-sigurd) [inline Modules/mathmodule.c]: Always put spaces around assignment, Boolean and comparison operators (PEP7).

File: Modules/mathmodule.c [hunk 3]
```diff
@@ -2031,6 +2031,87 @@ math_fmod_impl(PyObject *module, double x, double y)
...
+    }
+    if (found_nan) {
+        result = Py_NAN;
+        goto done;
+    }
+    if (max == 0.0) {
+        result = 0.0;
+        goto done;
+    }
+    for (i=0 ; i<n ; i++) {
```
Comment (sir-sigurd) [inline Modules/mathmodule.c:2106]: Always put spaces around assignment, Boolean and comparison operators (PEP7).

Comment (rhettinger) [reply Modules/mathmodule.c:2106]: Historically, we haven't done this inside for-loops.

File: Modules/mathmodule.c (hunk 2 above)
Comment (rhettinger) [reply Modules/mathmodule.c]: Historically, we haven't done this inside for-loops.

File: Modules/mathmodule.c (hunk 1 above)
Comment (rhettinger) [reply Modules/mathmodule.c]: Okay.

**Comment by LeonardAukea** [comment]
Hi, thanks for a great language. Not sure if this is the right place to add this comment. Anyways, was adding other norms ever under consideration?  For instance, `math.dist(p, q, norm='l1')`, where `euclidian` or `l2` would be default? I get that you may not want to add too many but at least the standard ones would be nifty.

**Pull request merged**